### PR TITLE
initial rework of v1.0.1

### PR DESCRIPTION
**Pull request by dnitsch**
refactored for tf12 only support

introduced the required configuration block as per the latest provider.

updated README and examples implementation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 
 # .tfvars files
 *.tfvars
+**/crash.log
```
Comment (ElvenSpellmaker) [inline .gitignore]: I'm not sure double star is supported, single should work for what you want, it ignores all crash logs in all folders?

File: README.md [hunk 2]
```diff
@@ -7,18 +7,39 @@ Terraform module which creates AKS Cluster resources on Azure.
...
   location            = "${var.resource_group_location}"
   resource_group_name = "${var.resource_group_name}"
   client_id           = "${var.cluster_spn_clientid}"
   client_secret       = "${var.cluster_spn_clientsecret}"
   tags                = "${var.resource_group_tags}"
-  
 }
 ```
 
+Always pin your module version using either the tagged release or sha_id of the version of tghe
```
Comment (ElvenSpellmaker) [inline README.md]: `the` not `tghe`

File: example/main.tf [hunk 3]
```diff
@@ -1,19 +1,71 @@
-provider "azurerm" {}
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  client_id       = var.client_id
+  client_secret   = var.client_secret
+  tenant_id       = var.tenant_id
+  version         = ">=1.38"
```
Comment (ElvenSpellmaker) [inline example/main.tf]: Should we pin < 2 as well?

File: example/main.tf [hunk 4]
```diff
@@ -1,19 +1,71 @@
...
+  name     = random_pet.name.id
+  location = var.resource_group_location
+
+  tags = {
+    environment = "example-test"
+  }
+}
+
+resource "azurerm_log_analytics_workspace" "default" {
+  name                = random_pet.name.id
```
Comment (ElvenSpellmaker) [inline example/main.tf]: Is this intention to choose a random pet name?

File: example/main.tf [hunk 5]
```diff
@@ -1,19 +1,71 @@
...
+
 
 resource "random_pet" "name" {
   length    = 2
   separator = ""
   prefix    = "aks"
 }
 
+resource "azurerm_resource_group" "default" {
+  name     = random_pet.name.id
```
Comment (ElvenSpellmaker) [inline example/main.tf]: Same here

File: example/main.tf [hunk 6]
```diff
@@ -1,19 +1,71 @@
...
-  cluster_name        = "${random_pet.name.id}"
-  agent_count         = "${var.cluster_agent_count}"
-  location            = "${var.resource_group_location}"
-  resource_group_name = "${random_pet.name.id}"
-  client_id           = "${var.cluster_spn_clientid}"
-  client_secret       = "${var.cluster_spn_clientsecret}"
-  tags                = "${var.resource_group_tags}"
+  # source = "git::https://github.com/amido/terraform-azure-akscluster.git=ref=v1.0.0"
+  source              = "../"
+  cluster_name        = "${random_pet.name.id}-#{REGION_ID}"
```
Comment (ElvenSpellmaker) [inline example/main.tf]: Pet name again, are you sure?

File: example/vars.tf [hunk 7]
```diff
@@ -1,10 +1,9 @@
 variable "resource_group_location" {
-  default = "northeurope"
+  default = "westeurope"
```
Comment (ElvenSpellmaker) [inline example/vars.tf]: Why are we changing the default here?

Comment (guibirow) [reply example/vars.tf]: if we are changing the default, maybe we should consider `uksouth`, probably will require a poll
I wouldn't change these defaults as part of this PR

File: README.md [hunk 8]
```diff
@@ -7,18 +7,39 @@ Terraform module which creates AKS Cluster resources on Azure.
 
 ```hcl
 module "aks-cluster" {
-  source = "git::https://github.com/amido/terraform-azure-akscluster.git"
+  source = "git::https://github.com/amido/terraform-azure-akscluster.git?ref=v1.0.0"
```
Comment (guibirow) [inline README.md]: if this is being tagged as `1.0.1`, shouldn't it be `1.0.1` as well?

File: example/main.tf [hunk 9]
```diff
@@ -1,19 +1,71 @@
...
   separator = ""
   prefix    = "aks"
 }
 
+resource "azurerm_resource_group" "default" {
+  name     = random_pet.name.id
+  location = var.resource_group_location
+
+  tags = {
+    environment = "example-test"
```
Comment (guibirow) [inline example/main.tf]: Do we need to set this default tag?
In some cases we might share the cluster as well, s o an environment tag might no apply

File: example/main.tf [hunk 10]
```diff
@@ -1,19 +1,71 @@
...
-  location            = "${var.resource_group_location}"
-  resource_group_name = "${random_pet.name.id}"
-  client_id           = "${var.cluster_spn_clientid}"
-  client_secret       = "${var.cluster_spn_clientsecret}"
-  tags                = "${var.resource_group_tags}"
+  # source = "git::https://github.com/amido/terraform-azure-akscluster.git=ref=v1.0.0"
+  source              = "../"
+  cluster_name        = "${random_pet.name.id}-#{REGION_ID}"
+  agent_count         = var.cluster_agent_count
+  cluster_version     = "1.14.8"
```
Comment (guibirow) [inline example/main.tf]: The cluster version should be passed as parameter

File: example/main.tf [hunk 11]
```diff
@@ -1,19 +1,71 @@
...
+    { enabled          = "true",
+      oms_workspace_id = azurerm_log_analytics_workspace.default.id
+    }
+  ]
+  # oms_ws_list_of_one  = []
+  resource_group_name = var.rg_name
+  client_id           = var.client_id
+  client_secret       = var.client_secret
+  tags                = var.resource_group_tags
+  agent_size          = "Standard_D3_v2"
```
Comment (guibirow) [inline example/main.tf]: This should also be passed as parameter

File: example/main.tf (hunk 3 above)
Comment (dnitsch) [reply example/main.tf]: this change will be >2 compatible since agent_pool is omitted and added to `lifecycle` `ignore_changes`

Comment (ElvenSpellmaker) [reply example/main.tf]: It's more, should we ever auto-upgrade to a major version?

**Pull request merged**